plugins/org.yakindu.sct.generator.core/src/org/yakindu/sct/generator/core/execution/AbstractGeneratorEntryExecutor.java

```diff
@@ -51,16 +55,23 @@
 	protected abstract void execute(ISCTFileSystemAccess access, GeneratorEntry generatorEntry);
 
 	@Override
-	public void execute(GeneratorEntry entry) {
+	public IStatus execute(GeneratorEntry entry) {
+		IStatus status = Status.OK_STATUS;
+		
 		logStart(entry);
 		if (valid(entry)) {
 			try {
 				execute(factory.create(entry), entry);
 			} catch (Exception ex) {
 				logger.logError(ex);
+				status = new Status(IStatus.ERROR, PLUGIN_ID, ex.getMessage());
 			}
+		} else {
+			status = new Status(IStatus.ERROR, PLUGIN_ID,
```

Comment:
is Status.ERROR correct here since there is no error occured during generation. Maybe Status.Warning or Status.Cancel is betteR?

Reply:
At this point the model validation has found an error in the referenced statechart. These errors are also already printed out on console log. So it is a validation error here.